jaxrs/src/main/java/org/jboss/as/jaxrs/deployment/JaxrsScanningProcessor.java

```diff
@@ -197,13 +197,13 @@ protected boolean hasBootClasses(JBossWebMetaData webdata) {
         if (webdata.getServlets() != null) {
             for (ServletMetaData servlet : webdata.getServlets()) {
                 String servletClass = servlet.getServletClass();
-                if (BOOT_CLASSES.contains(servletClass))
+                if (servletClass != null && BOOT_CLASSES.contains(servletClass))
                     return true;
             }
         }
         if (webdata.getFilters() != null) {
             for (FilterMetaData filter : webdata.getFilters()) {
-                if (BOOT_CLASSES.contains(filter.getFilterClass()))
+                if (filter != null && BOOT_CLASSES.contains(filter.getFilterClass()))
```

Comment:
Should this also check if filter.getFilterClass()) returns null?

Reply:
Oops, yes. Nice catch.